### DISCUSSION TITLE
Viewing Archived Workflows when RBAC is Namespace Delegated - Bug?

### DISCUSSION THREAD

**Discussion by BhavikaSharma** [argoproj/argo-workflows#8497] [Q&A] [answered]
Hi, I'm wondering if there's a potential bug when when using namespace delegation for RBAC. I have a service account in a namespace that has permissions to view workflows in the namespace. I'm able to log in, and through a group, bind to the service account. This allows me to view workflows in the namespace.

From the logs, I see:

```
time="2022-04-26T20:53:26.926Z" level=info msg="selected SSO RBAC service account for user" email=<redacted> loginServiceAccount=user-default-login serviceAccount=default ssoDelegated=true ssoDelegationAllowed=true subject=<redacted>
```

However, when accessing archived workflows from the UI, I receive an error:
```
time="2022-04-26T17:54:01.366Z" level=info msg="finished unary call with code OK" grpc.code=OK grpc.method=GetInfo grpc.service=info.InfoService grpc.start_time="2022-04-26T17:54:01Z" grpc.time_ms=2.549 span.kind=server system=grpc
time="2022-04-26T17:54:01.376Z" level=warning msg="finished unary call with code PermissionDenied" error="rpc error: code = PermissionDenied desc = Permission denied, you are not allowed to list workflows in namespace \"<redacted>\". Maybe you want to specify a namespace with `listOptions.fieldSelector=metadata.namespace=your-ns`?" grpc.code=PermissionDenied grpc.method=ListArchivedWorkflows grpc.service=workflowarchive.ArchivedWorkflowService grpc.start_time="2022-04-26T17:54:01Z" grpc.time_ms=15.412 span.kind=server system=grpc
```

From the logs, I see:

```
time="2022-04-26T20:52:30.920Z" level=info msg="selected SSO RBAC service account for user" email=<redacted> loginServiceAccount=user-default-login serviceAccount=user-default-login ssoDelegated=false ssoDelegationAllowed=false subject=<redacted>
```
When I expected to see the same logs as regularly accessing the workflows page. 

It seems like [this](https://github.com/argoproj/argo-workflows/blob/e12827b8b0ecb11425399608b1feee2ad739575d/server/auth/gatekeeper.go#L274) function is returning false for archived workflows. Wondering if anyone else is seeing this/if this is a potential bug. Thanks!

**Comment by alexec** [comment, marked as answer]
This seems like a bug to me. Could you please raise an issue?

**Reply by BhavikaSharma** [reply]
Will do

**Comment by fouadsemaan** [comment]
@BhavikaSharma Can you please share your Argoserver config and in particular the SSO config?
